pipeline/metadata/transmission/windowPlot.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python3
+
+import ROOT, math
+
+outfname = "windowsTransmission.png"
+
+from ROOT import (
+     TChain, TFile, TTree, TCanvas, TPad, TRandom3,
+     TH1D, TH2D, TH3D,
+     TProfile, TProfile2D, TProfile3D,
+     TGraph, TGraph2D,
+     TF1, TF2, TF3, TFormula,
+     TLorentzVector, TVector3)
+
+ROOT.gSystem.Load("libRestFramework.so")
+ROOT.gSystem.Load("libRestAxion.so")
+
+### Creating a canvas and pad for drawing
+c1 = TCanvas( 'c1', 'My canvas', 1200,400 )
+c1.GetFrame().SetBorderSize( 6 )
+c1.GetFrame().SetBorderMode( -1 )
+
+pad1 = TPad("pad1","This is pad1",0.01,0.02,0.99,0.97);
+pad1.Divide(3,1)
+pad1.Draw()
+
+totalSamples = 100000
+
+cathode = ROOT.TRestAxionXrayWindow("windows.rml", "cathode")
+strongBack = ROOT.TRestAxionXrayWindow("windows.rml", "strongBack")
+siFoil = ROOT.TRestAxionXrayWindow("windows.rml", "siliconFoil")
+
+radius = cathode.GetWindowRadius()
+print ( "\nGetting window radius" )
+if( radius != 8 ):
+    print ("\nThe window radius is not as expected! Exit code : 102")
+    exit(102)
+print ( "[\033[92m OK \x1b[0m]" )
+
+# L: Low energy (0-5)keV M: Medium energy H: High energy
+histL = ROOT.TH2D("Low", "Low energy", 80, -radius-2, radius+2, 80, -radius-2, radius+2 )
+histM = ROOT.TH2D("Medium", "Medium energy", 80, -radius-2, radius+2, 80, -radius-2, radius+2 )
+histH = ROOT.TH2D("High", "High energy", 80, -radius-2, radius+2, 80, -radius-2, radius+2 )
+
+#for n in range(rings):
+#    gr = ROOT.TGraph()
+#    gr.SetMarkerStyle(20)
+#    gr.SetMarkerSize(0.5)
+#    gr.SetMarkerColor(38+n)
+#    gr.SetTitle("Optics plane (no-Spider)")
+#    gr.GetXaxis().SetTitle("X [mm]")
+#    gr.GetXaxis().SetTitleSize(0.05);
+#    gr.GetXaxis().SetLabelSize(0.05);
+#    gr.GetYaxis().SetTitle("Y [mm]")
+#    gr.GetYaxis().SetTitleOffset(1)
+#    gr.GetYaxis().SetTitleSize(0.05);
+#    gr.GetYaxis().SetLabelSize(0.05);
+#    graphsOP.append(gr)
+
+
+rnd = TRandom3(0)
```

Comment:
You are right. We do fix the seed value on event processing that needs to produce always the same results for validation pourposes. But in this case I believe is indifferent.

Reply:
Even if in this case we don't currently use it as a direct validation, I still think a reproducible result is preferable.

Reply:
I would prefer to calculate the radius average for the points falling in a particular ring and check the radius is at the appropiate ring rather than comparing pngs.

That would be also independent from `TRandom` generation, and from any `png` library or `ROOT` library update that, let's imagine, changes the color of the pixel, or the default font, or whatever else which is irrelevant to our routine.

Reply:
This validation for the moment just checks the plot is really generated (i.e. the script gets to the end without problems). Further tests may be added in the future.

Reply:
> I would prefer to calculate the radius average for the points falling in a particular ring and check the radius is at the appropriate ring rather than comparing png images.

I'm not proposing to compare PNG images (even though I do it [here](https://github.com/Vindaar/ggplotnim/blob/master/tests/tCompareRecipes.nim#L59) myself :hand_over_mouth: ). Your proposal sounds like a decent validation.

But I'm saying that a validation pipeline shouldn't introduce variability where not required. Simply choose a fixed seed. In the worst case you never notice it. In the best case it avoids you or someone else at some point debugging why something changes each run (e.g. someone extends some pipeline etc.). Also the `0` is kind of a magic number, that *appears* to be a fixed seed when it isn't. That's misleading in addition to that.

And still, given that the plot is committed, it will lead to each run causing the plot to appear in unstaged changes, which is simply annoying.

Reply:
Then, I reply you with my first answer yesterday and we enter in an infinite loop.